crates/noirc_errors/src/debug_info.rs

```diff
@@ -29,16 +29,14 @@ impl DebugInfo {
     /// renders the old `OpcodeLocation`s invalid. The AcirTransformationMap is able to map the old `OpcodeLocation` to the new ones.
     /// Note: One old `OpcodeLocation` might have transformed into more than one new `OpcodeLocation`.
     pub fn update_acir(&mut self, update_map: AcirTransformationMap) {
-        let mut new_locations_map = BTreeMap::new();
+        let old_locations = self.locations.clone();
+        self.locations.clear();
 
-        for (old_opcode_location, source_locations) in &self.locations {
-            let new_opcode_locations = update_map.new_locations(*old_opcode_location);
-            for new_opcode_location in new_opcode_locations {
-                new_locations_map.insert(new_opcode_location, source_locations.clone());
-            }
+        for (old_opcode_location, source_locations) in old_locations {
+            let _ = update_map.new_locations(old_opcode_location).map(|new_opcode_location| {
+                self.locations.insert(new_opcode_location, source_locations.clone());
```

Comment:
We're doing a lot of cloning of BTreeMaps, can we avoid this?

Reply:
Nevermind, these are Vecs. We could experiement with `im::Vector` perhaps to see if cloning those are any faster.

Reply:
Also, `Vec` is not an iterator so that's why the compiler error when calling `map` is happening. Also note that without calling another function afterward, `map` is never actually run so `self.locations` will be empty. You're looking for `for_each` or just a for loop instead.

It also looks like in `new_locations` we're collecting into a Vec only to immediately discard it by iterating over it. We could experiment with changing `new_locations` to return an iterator directly instead. Or it may be easier to have it accept a function as an argument to apply to each element.